### PR TITLE
Migrate repo to WPILib 2021.1.2

### PR DESCRIPTION
**Pull request by FireShredder24**

### CODE REVIEW

**Review by terbos90803 — COMMENTED** [review]
This all looks reasonable.

Does anyone understand why some of the diffs don't appear to change the contents of the file, yet the whole file is flagged?  I don't see any visible changes to spacing or anything.

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -176,7 +176,8 @@ private void configureButtonBindings() {
    * @return the command to run in autonomous
    */
   public Command getAutonomousCommand() {
-    return autoChooser.getSelected();
```
Comment (jordan-powers) [inline src/main/java/frc/robot/RobotContainer.java]: Is this part of the migration, or is this an out-of-scope change which slipped in?

**Comment by jordan-powers** [comment]
I'm guessing the line endings changed. That would cause the diffs appear to show the whole file changing, with no actual substantial changes.

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (FireShredder24) [reply src/main/java/frc/robot/RobotContainer.java]: I included this out of scope to test the autonomous command.  I didn't intend for it to be pushed at all, but apparently I didn't clean the working tree before committing.

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
-    "**/CVS": true,
-    "**/.DS_Store": true,
-    "bin/": true,
-    "**/.classpath": true,
-    "**/.project": true,
-    "**/.settings": true,
-    "**/.factorypath": true
-  },
-  "editor.tabSize": 2,
-  "editor.insertSpaces": true,
```
Comment (terbos90803) [inline .vscode/settings.json:16]: I think we should retain these editor settings.  Makes for happier merges.

**Review by terbos90803 — CHANGES_REQUESTED** [review]

**Review by terbos90803 — APPROVED** [review]
Looks much better.

**Pull request merged**